includes/gateway/class-omise-payment-creditcard.php

```diff
@@ -166,243 +166,192 @@ public function payment_fields() {
-								'order' => 'reverse_chronological'
-							) );
-
-							$card_id = $cards['data'][0]['id'];
-						} catch (Exception $e) {
-							throw new Exception( $e->getMessage() );
-						}
-					} else {
-						$description   = "WooCommerce customer " . $user->ID;
-						$customer_data = array(
-							"description" => $description,
-							"card"        => $token
-						);
-
-						$omise_customer = OmiseCustomer::create( $customer_data );
-
-						if ( $omise_customer['object'] == "error" ) {
-							throw new Exception( $omise_customer['message'] );
-						}
-
-						$omise_customer_id = $omise_customer['id'];
-						if ( $this->is_test() ) {
-							update_user_meta( $user->ID, 'test_omise_customer_id', $omise_customer_id );
-						} else {
-							update_user_meta( $user->ID, 'live_omise_customer_id', $omise_customer_id );
-						}
-
-						if ( 0 == sizeof( $omise_customer['cards']['data'] ) ) {
-							throw new Exception(
-								sprintf(
-									wp_kses(
-										__( 'Note that nothing wrong by you, this might be from our store issue.<br/><br/>Please feel free to try submit your order again or report our support team that you have found this problem (Your temporary order id is \'%s\')', 'omise' ),
-										array(
-											'br' => array()
-										)
-									),
-									$order_id
-								)
-							);
-						}
+					$omise_customer = OmiseCustomer::create( $customer_data );
 
-						$cards   = $omise_customer->cards( array( 'order' => 'reverse_chronological' ) );
-						$card_id = $cards['data'][0]['id']; //use the latest card
+					if ( $omise_customer['object'] == "error" ) {
+						throw new Exception( $omise_customer['message'] );
 					}
-				}
 
-				$success = false;
-				$data    = array(
-					'amount'      => $this->format_amount_subunit( $order->get_total(), $order->get_order_currency() ),
-					'currency'    => $order->get_order_currency(),
-					'description' => apply_filters( 'omise_charge_params_description', 'WooCommerce Order id ' . $order_id, $order ),
-					'return_uri'  => add_query_arg( 'order_id', $order_id, site_url() . '?wc-api=omise_callback' )
-				);
+					$omise_customer_id = $omise_customer['id'];
+					if ( $this->is_test() ) {
+						update_user_meta( $user->ID, 'test_omise_customer_id', $omise_customer_id );
+					} else {
+						update_user_meta( $user->ID, 'live_omise_customer_id', $omise_customer_id );
+					}
 
-				if ( ! empty( $omise_customer_id ) && ! empty( $card_id ) ) {
-					$data['customer'] = $omise_customer_id;
-					$data['card']     = $card_id;
-				} else {
-					$data['card'] = $token;
-				}
+					if ( 0 == sizeof( $omise_customer['cards']['data'] ) ) {
+						throw new Exception(
+							sprintf(
+								wp_kses(
+									__( 'Note that nothing wrong by you, this might be from our store issue.<br/><br/>Please feel free to try submit your order again or report our support team that you have found this problem (Your temporary order id is \'%s\')', 'omise' ),
```

Comment:
Please note that you've done nothing wrong - this is likely an issue with our store.<br/><br/>Feel free to try submitting your order again, or report to our support team that you have found this problem (Your temporary order id is